src/eventHubManager.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { EventData, EventHubClient, EventPosition, MessagingError, OnError, OnMessage } from "@azure/event-hubs";
+import EventHubManagementClient from "azure-arm-eventhub";
+import * as vscode from "vscode";
+import { Constants } from "./constants";
+import { IoTHubMessageBaseExplorer } from "./iotHubMessageBaseExplorer";
+import { EventHubItem } from "./Model/EventHubItem";
+import { TelemetryClient } from "./telemetryClient";
+import { Utility } from "./utility";
+
+export class EventHubManager extends IoTHubMessageBaseExplorer {
+    private _eventHubClient: EventHubClient;
+
+    constructor(outputChannel: vscode.OutputChannel) {
+        super(outputChannel, "$(primitive-square) Stop Monitoring Custom Event Hub Endpoint", "azure-iot-toolkit.stopMonitorCustomEventHubEndpoint");
+    }
+
+    public async startMonitorCustomEventHubEndpoint(eventHubItem: EventHubItem) {
+        if (this._isMonitoring) {
+            this._outputChannel.show();
+            this.outputLine(Constants.IoTHubMonitorLabel, "There is a running job to monitor custom Event Hub endpoint. Please stop it first.");
+            return;
+        }
+
+        try {
+            TelemetryClient.sendEvent(Constants.IoTHubAIEHStartMonitorEvent);
+            this._outputChannel.show();
+            this.outputLine(Constants.EventHubMonitorLabel, `Start monitoring message arrived in custom Event Hub endpoint [${eventHubItem.eventHubProperty.name}] ...`);
+
+            const eventHubClient = new EventHubManagementClient(eventHubItem.azureSubscription.session.credentials, eventHubItem.azureSubscription.subscription.subscriptionId);
+            const connectionString = (await eventHubClient.namespaces.listKeys(eventHubItem.eventHubProperty.resourceGroup,
+                this.getNamespacefromConnectionString(eventHubItem.eventHubProperty.connectionString), "RootManageSharedAccessKey")).primaryConnectionString;
```

Comment:
So to monitor the message must have "RootManageSharedAccessKey" role?

Reply:
Yes, it is similar to the concept of 'iothubowner' in IoT Hub. Though user could add his own, but we use this directly.